controllers/state_of_the_world.go

```diff
@@ -64,14 +72,137 @@ func NewPolicyMachineryController(manager ctrlruntime.Manager, client *dynamic.D
+			},
+		}).Run,
 		Tasks: []controller.ReconcileFunc{
-			NewTopologyFileReconciler(client, operatorNamespace).Reconcile,
+			NewAuthorinoCrReconciler(client).Subscription().Reconcile,
 		},
 	}
 	return reconciler.Run
 }
 
+type AuthorinoCrReconciler struct {
+	Client *dynamic.DynamicClient
+}
+
+func NewAuthorinoCrReconciler(client *dynamic.DynamicClient) *AuthorinoCrReconciler {
+	return &AuthorinoCrReconciler{Client: client}
+}
+
+func (r *AuthorinoCrReconciler) Subscription() *controller.Subscription {
+	return &controller.Subscription{
+		ReconcileFunc: r.Reconcile,
+		Events: []controller.ResourceEventMatcher{
+			{Kind: ptr.To(kuadrantv1beta1.KuadrantKind), EventType: ptr.To(controller.CreateEvent)},
+			{Kind: ptr.To(kuadrantv1beta1.AuthorinoKind), EventType: ptr.To(controller.DeleteEvent)},
+		},
+	}
+}
+
+func (r *AuthorinoCrReconciler) Reconcile(ctx context.Context, events []controller.ResourceEvent, topology *machinery.Topology, _ error) {
+	logger := controller.LoggerFromContext(ctx).WithName("AuthorinoCrReconciler")
+	logger.Info("Reconciling Authorino Cr")
+
+	kobj := &kuadrantv1beta1.Kuadrant{}
+	for _, event := range events {
+		if event.Kind == kuadrantv1beta1.KuadrantKind && event.EventType == controller.CreateEvent {
+			kobjs := lo.FilterMap(topology.Objects().Roots(), func(item machinery.Object, _ int) (*kuadrantv1beta1.Kuadrant, bool) {
+				if item.GetName() == event.NewObject.GetName() && item.GetNamespace() == event.NewObject.GetNamespace() && item.GroupVersionKind().Kind == event.NewObject.GetObjectKind().GroupVersionKind().Kind {
+					return item.(*kuadrantv1beta1.Kuadrant), true
+				}
+				return nil, false
+			})
+			if len(kobjs) != 1 {
+				logger.Error(fmt.Errorf("muiltply kuadrant CRs found"), "unexpected behaviour may happen")
+			}
+			kobj = kobjs[0]
+			break
+		} else if event.Kind == kuadrantv1beta1.AuthorinoKind && event.EventType == controller.DeleteEvent {
+			kobjs := lo.FilterMap(topology.Objects().Roots(), func(item machinery.Object, _ int) (*kuadrantv1beta1.Kuadrant, bool) {
+				if item.GetNamespace() == event.OldObject.GetNamespace() && item.GroupVersionKind().Kind == kuadrantv1beta1.KuadrantKind.Kind {
+					return item.(*kuadrantv1beta1.Kuadrant), true
+				}
+				return nil, false
+			})
+
+			if len(kobjs) == 0 {
+				logger.Info("no possible kuadrant parent, wont create Authorino CR.")
+				return
+			}
+			if len(kobjs) != 1 {
+				logger.Error(fmt.Errorf("muiltply kuadrant CRs found"), "unexpected behaviour may happen")
+			}
+			kobj = kobjs[0]
+			if kobj.GetDeletionTimestamp() != nil {
+				logger.Info("kuadrant CR marked for deletion, wont create Authorino CR.")
+				return
+			}
+			break
+		}
+	}
+
+	authorino := &authorinov1beta1.Authorino{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Authorino",
+			APIVersion: "operator.authorino.kuadrant.io/v1beta1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "authorino",
+			Namespace: kobj.Namespace,
```

Comment:
If I understand the selection of the Kuadrant object above correctly, if a second Kuadrant CR is created (in another namespace), another Authorino instance will be created for it too. Is that what we want? I think the only way to support this is by also supporting sharded instances (#274).

If a second Kuadrant CR is created in the same namespace where another exists, then we know we won't create another instance of Authorino in that same namespace (with static name `authorino`), right? So why even continue if don't reconcile existing Authorino CRs? And if we do decide to reconcile those, as soon as it starts having any field set directly from the Kuadrant CR, then the current code could trigger an infinite loop of reconciliation events with 2+ Kuadrant CRs.